.pre-commit-config.yaml

```diff
@@ -6,7 +6,8 @@ repos:
     rev: v0.8.8
     hooks:
       - id: licenseheaders
-        args: ["--tmpl=LICENSE_SHORT", "--ext=py", "-f"]
+        args:
+          ["--tmpl=LICENSE_SHORT", "--ext=py", "-x=**/models/__init__.py", "-f"]
```

Comment:
That's very interesting and yeah ideally we would just have one source of truth. Maybe something to look into. What is curious though is why are the Airbyte protocol models and `__init__` files not affected by this because presumably it was working for that as well? Or is it because we don't currently perform this diff check for those files?

Reply:
I agree that it seems like it would make sense to use just one formatting pipeline but am also not sure if there's a reason for the discrepancy.

Were you getting `pre-commit` failures if you didn't exclude `**/models/__init__.py` here?

Reply:
> What is curious though is why are the Airbyte protocol models and __init__ files not affected by this because presumably it was working for that as well?

I was wondering. It comes for [this PR](https://github.com/airbytehq/airbyte/pull/3388). Apparently, this is when the tool that adds the license was added. Why does it exclude the `__init__.py` files at that stage though? I'm not sure. This is interesting because pre-commit does not exclude the `__init__.py` files so we end up having the licence on `__init__.py` files anyway.

My confusion probably comes from the fact that https://github.com/airbytehq/airbyte/blob/master/airbyte-cdk/python/airbyte_cdk/models/__init__.py does not have the license headers. My guess is that it should have those headers so I'll remove this exclusion from pre-commit and update this file.

> Were you getting pre-commit failures if you didn't exclude **/models/__init__.py here?

Yes because pre-commit expect the license to be there on all python files (which probably makes sense so I'll remove this exclusion)

Reply:
After removing the `-x=**/models/__init__.py` from `licenseheaders`, there is an weird drawback where when I run `SUB_BUILD=CONNECTORS_BASE ./gradlew format --scan --info --stacktrace` (the command that should generate and format the autogenerate files), I end up with a diff where the new generated `__init__.py` files don't have the license header. This is not so great because:
* There were actually no change to this file except the license being removed
* pre-commit will re-add the license before commiting which will bring back the file as it was before the `./gradlew format`
My conclusion is that we can't change only one without changing both. If we want to stop excluding `__init__.py` files from the `./gradlew format`, it will update 42 files currently under airbyte-cdk

Reply:
Okay thanks for really digging into this! I think its okay to retain the exclusion of `__init__`, this is pretty minimal and for an autogenerated file.

Out of curiosity, what types of files were the 42 files? If these were legitimate failures of us forgetting a license header or something, might be worth a follow up ticket or something

Reply:
It all seems very minor to me:
* Copyright date is updated from 2021 to 2022
* Header was completely missing and is added
* Breakline was missing at the end of the license